common-rest/src/main/java/org/uniprot/api/rest/service/DefaultSearchQueryOptimiser.java

```diff
@@ -0,0 +1,105 @@
+package org.uniprot.api.rest.service;
+
+import static org.uniprot.core.util.Utils.notNullNotEmpty;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.lucene.analysis.core.WhitespaceAnalyzer;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.uniprot.api.common.exception.ServiceException;
+import org.uniprot.store.config.searchfield.model.SearchFieldItem;
+
+/**
+ * @author lgonzales
+ * @since 10/06/2020
+ */
+public class DefaultSearchQueryOptimiser {
+
+    private final List<SearchFieldItem> optimisedFields;
+
+    public DefaultSearchQueryOptimiser(SearchFieldItem idField) {
+        this(Collections.singletonList(idField));
+    }
+
+    public DefaultSearchQueryOptimiser(List<SearchFieldItem> optimisedFields) {
+        this.optimisedFields = optimisedFields;
```

Comment:
call it `optimisableFields` or `fieldsToOptimise`, so it doesn't sound like these fields are already optimised?